standards/ICRC-3/README.md

```diff
@@ -0,0 +1,179 @@
+     };
+     timestamp : nat64;
+};
+```
+
+The client specifies the index of the first transaction and the number of transactions to fetch.
+
+The ledger returns a record with the following fields:
+  * The `log_length` field is the total number of transactions in the log.
+  * The `transactions` field is an _infix_ of the requested transaction range.
+  * The `first_index` field is the index of the first transaction in the `transaction` field.
+    If the `transactions` field is an empty vector, the value of `first_index` is unspecified.
+  * The `archived_transactions` field contains instructions for fetching the _prefix_ of the requested range.
+    Each entry indicates that the client can fetch transactions in the range `[start, length]` with the specified `callback` method reference.
+
+```candid "Type definitions" +=
+type GetTransactionsRequest = record { start : TxIndex; length : nat };
+
+type GetTransactionsResponse = record {
+    log_length : nat;
+    transactions : vec Transaction;
+    first_index : TxIndex;
+    archived_transactions : vec record {
+        start : TxIndex;
+        length : nat;
+        callback : QueryArchiveFn;
+    };
+};
+```
+
+Some of the transactions in the range might be "archived", i.e., reside in other canisters.
+All transaction ranges that `{ start; length }` tuples in the `archived_transactions` field form MUST have a non-zero intersection with the requested range.
+
+```candid "Type definitions" +=
+type QueryArchiveFn = func (GetTransactionsRequest) -> (TransactionRange) query;
+
+type TransactionRange = record { transactions : vec Transaction; };
+```
+
+Ledger and archives MAY return fewer transactions than the client requested.
+
+## Examples
+
+### Synchronizing the state
+
+The following example demonstrates how a client can synchronize with the ledger state distributed across multiple canisters using the proposed interface.
+
+  1. The client calls `icrc3_get_transactions({ start = 0; length = 10_000 })` on the ledger.
+  2. The ledger has 9_500 transactions and happens to have transactions `4_000..5_500` in memory.
+     However, the ledger implementors decided not to return more than 1_000 per request.
+  3. The ledger returns the following value.
+     ```candid
+     record {
+        log_length = 5_500;
+        transactions = vec { /* transactions 4_000..5_000 */ };
+        first_index = 4_000;
+        archived_transactions = vec {
+            record { start = 0; length = 4_000; callback = "4kydj-ryaaa-aaaag-qaf7a-cai"."get_archived_transactions" }
+        }
+     }
+     ```
+  4. The client appends transactions `4_000..5_000` to the local buffer and issues a follow-up call to the archive: `get_archived_transactions({ start = 0; length = 4_000 })`.
+  5. The archive implementors decided not to return more than 2_000 transactions per request.
+     The archive returns the following value.
+     ```candid
+     record { transactions : vec { /* transactions 0..2_000 */ } }
```

Comment:
off-by-1 error: should probably be { /* transactions 0..1_999 */ } as only 2000 trx can be returned
analogous below for the remaining part of the example

Reply:
I use Rust syntax for ranges here, which does not include the right bound: `0..2000` means `[0, 2000)` in standard math notation. We can also switch to the math notation to avoid confusion.

Reply:
Ah, OK. I guess if we make it clear, it should be fine as well. But otherwise, it is confusing. I proposed a change to standard notation, feel free to reject it and make clear what kind of notation it is. Not sure what is less confusing to the majority of people.